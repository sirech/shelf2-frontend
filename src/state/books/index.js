import * as constants from './constants'
import * as actions from './actions'

export { actions, constants }
export { default as reducer } from './reducers'
