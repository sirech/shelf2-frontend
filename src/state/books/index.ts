import * as constants from './constants'
import * as actions from './slice'

export { actions, constants }
export { default as booksSelector } from './selectors'
export { default as reducer } from './slice'
