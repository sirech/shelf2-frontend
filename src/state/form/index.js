import * as constants from './constants'
import * as actions from './actions'

export { constants, actions }
export { default as formSelector, bookSelector } from './selectors'
export { default as reducer } from './reducers'

export { modelName } from './utils'
