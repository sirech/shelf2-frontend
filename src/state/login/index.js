import * as constants from './constants'
import * as actions from './actions'

export { constants, actions }
export { default as loginSelector } from './selectors'
export { default as reducer } from './reducers'
