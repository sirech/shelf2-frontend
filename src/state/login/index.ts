import * as constants from './constants'
import * as actions from './actions'

export { constants, actions }
export {
  default as loginSelector,
  authenticatedSelector,
  failedSelector,
} from './selectors'
export { default as reducer } from './reducers'
