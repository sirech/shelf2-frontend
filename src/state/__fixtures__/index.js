import R from 'ramda'

import books from './books'
import years from './years'
import form from './form'
import login from './login'

const modal = () => ({
  opened: false
})

const search = R.pipe(books, R.omit(['activeYear']))

const state = () => {
  return {
    books: books(),
    search: search(),
    years: years(),
    form: form(),
    modal: modal(),
    login: login()
  }
}

export { books, search, years, form, login }
export default state
