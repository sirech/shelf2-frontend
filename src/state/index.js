import { combineReducers } from 'redux'

import * as R from 'ramda'

import { constants as booksConstants, reducer as booksReducer } from './books'
import {
  constants as searchConstants,
  reducer as searchReducer,
} from './search'
import { constants as yearsConstants, reducer as yearsReducer } from './years'
import { constants as formConstants, reducer as formReducer } from './form'
import { constants as modalConstants, reducer as modalReducer } from './modal'
import { constants as loginConstants, reducer as loginReducer } from './login'

export const actionPicker = list => {
  return actions => {
    const h = R.pick(list)(actions)

    if (Object.keys(h).length !== list.length) {
      throw new Error(
        `Not all requested actions (${list}) exist in the given actions`
      )
    }

    return h
  }
}

const rootReducer = combineReducers({
  [booksConstants.namespace]: booksReducer,
  [searchConstants.namespace]: searchReducer,
  [yearsConstants.namespace]: yearsReducer,
  [formConstants.namespace]: formReducer,
  [modalConstants.namespace]: modalReducer,
  [loginConstants.namespace]: loginReducer,
})

export default rootReducer
