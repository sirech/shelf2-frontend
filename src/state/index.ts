import { ThunkAction, UnknownAction, configureStore } from '@reduxjs/toolkit'

import { constants as booksConstants, reducer as booksReducer } from './books'
import {
  constants as searchConstants,
  reducer as searchReducer,
} from './search'
import { constants as yearsConstants, reducer as yearsReducer } from './years'
import { constants as loginConstants, reducer as loginReducer } from './login'

const store = configureStore({
  reducer: {
    [booksConstants.namespace]: booksReducer,
    [searchConstants.namespace]: searchReducer,
    [yearsConstants.namespace]: yearsReducer,
    [loginConstants.namespace]: loginReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  UnknownAction
>

export default store
