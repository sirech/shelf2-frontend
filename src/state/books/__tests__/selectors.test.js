import state from '../../__fixtures__'
import booksSelector from '../selectors'

describe('booksSelector', () => {
  it('selects the books', () => {
    expect(booksSelector(state())).toMatchSnapshot()
  })
})
