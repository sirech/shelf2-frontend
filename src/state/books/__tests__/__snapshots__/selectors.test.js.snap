// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`booksSelector selects the books 1`] = `
Object {
  "entities": Object {
    "books": Object {
      "10": Object {
        "category": "other",
        "id": 10,
        "stars": 1,
        "title": "Carrion Comfort",
        "year": 2016,
      },
      "11": Object {
        "category": "history",
        "id": 11,
        "stars": 2,
        "title": "Time To Murder And Create",
        "year": 2016,
      },
      "4": Object {
        "category": "history",
        "id": 4,
        "stars": 5,
        "title": "Fear and Trembling",
        "year": 2016,
      },
      "6": Object {
        "category": "sociology",
        "id": 6,
        "stars": 1,
        "title": "The Painted Veil",
        "year": 2016,
      },
      "7": Object {
        "category": "other",
        "id": 7,
        "stars": 1,
        "title": "Things Fall Apart",
        "year": 2016,
      },
      "8": Object {
        "category": "sociology",
        "id": 8,
        "stars": 2,
        "title": "Alone on a Wide, Wide Sea",
        "year": 2016,
      },
      "9": Object {
        "category": "other",
        "id": 9,
        "stars": 2,
        "title": "I Know Why the Caged Bird Sings",
        "year": 2016,
      },
    },
  },
  "result": Array [
    8,
    10,
    4,
    9,
    6,
    7,
    11,
  ],
}
`;
